testing/test_capture.py

```diff
@@ -1402,30 +1402,42 @@ def test_capattr():
     reprec.assertoutcome(passed=1)
 
 
+@pytest.mark.skipif("sys.version_info >= (3,)", reason="only relevant for py2")
 def test_crash_on_closing_tmpfile_py27(testdir):
-    testdir.makepyfile(
+    p = testdir.makepyfile(
         """
         from __future__ import print_function
         import time
         import threading
         import sys
 
+        printing = threading.Event()
+
         def spam():
+            global printing
+
             f = sys.stderr
+            printing.set()
             while True:
-                print('.', end='', file=f)
+                print('SPAM', end='', file=f)
+                f.flush()
 
         def test_silly():
+            global printing
+
             t = threading.Thread(target=spam)
             t.daemon = True
             t.start()
-            time.sleep(0.5)
 
+            while not printing.isSet():
```

Comment:
Perhaps `printing.wait()` instead?

Reply:
Hmm noticed this failed on Linux. You might try to only set the event after printing a number of times?

Reply:
For reference: all failed with `AssertionError: assert 'SPAMSPAMSPAM' in ''`.

Pushed a fixup based on your suggestions.